### PR TITLE
[DSM] PEPPER-798, PEPPER-1204 sm id, tissue and onc history delete fix

### PR DESCRIPTION
**Pull request by pegahtah**
I have added small fixes here and there, very minimal with smallest changes possible to get the delete working

### CODE REVIEW

File: ddp-workspace/projects/ddp-dsm-ui/src/app/onc-history-detail/onc-history-detail.model.ts [hunk 1]
```diff
@@ -55,12 +55,17 @@ export class OncHistoryDetail {
   }
 
   static parse(json: OncHistoryDetailsInterface): OncHistoryDetail {
+    if (json.deleted) {
+      return null;
+    }
```
Comment (pegahtah) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/onc-history-detail/onc-history-detail.model.ts:60]: don't show deleted data in participant list by stopping the parsing

File: ddp-workspace/projects/ddp-dsm-ui/src/app/onc-history-detail/onc-history-detail.model.ts [hunk 2]
```diff
@@ -55,12 +55,17 @@ export class OncHistoryDetail {
...
     const tissues: Array<Tissue> = [];
     const jsonData = json.tissues;
     if (jsonData != null) {
       jsonData.forEach((val) => {
         const tissue = Tissue.parse(val);
-        tissues.push(tissue);
+        if (tissue) {
+          tissues.push(tissue);
+        }
```
Comment (pegahtah) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/onc-history-detail/onc-history-detail.model.ts:68]: if tissue is deleted the parse will return null and so don't push it to tissues

File: ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/participant-list.model.ts [hunk 3]
```diff
@@ -39,7 +39,9 @@ export class Participant {
     if (jsonData != null) {
       jsonData.forEach((val) => {
         const oncHistory = OncHistoryDetail.parse(val);
-        oncHistoryDetails.push(oncHistory);
+        if (oncHistory) {
+          oncHistoryDetails.push(oncHistory);
```
Comment (pegahtah) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/participant-list.model.ts:43]: if onc history is deleted the parse will return null and so don't add that in the array and don't push it

File: ddp-workspace/projects/ddp-dsm-ui/src/app/tissue/tissue.component.ts [hunk 4]
```diff
@@ -181,6 +182,12 @@ export class TissueComponent {
           if (data['smIdPk']) {
             smIdArray[index].smIdPk = data['smIdPk'];
           }
+          if (tAlias === 'sm') {
+            if (this.smIdDuplicate[this.currentSMIDField].has(this.createDuplicateIndex(index))) {
+              this.smIdDuplicate[this.currentSMIDField].delete(this.createDuplicateIndex(index));
```
Comment (pegahtah) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/tissue/tissue.component.ts:191]: if a duplicate sm value is fixed, remove it's error

File: ddp-workspace/projects/ddp-dsm-ui/src/app/tissue/tissue.component.ts [hunk 5]
```diff
@@ -193,13 +200,17 @@ export class TissueComponent {
...
+          result= false;
         },
       });
     }
+    return result;
   }
 
   deleteTissue(): void {
-    this.tissue.deleted = true;
+    if (this.valueChanged('1', 'deleted')){
```
Comment (pegahtah) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/tissue/tissue.component.ts]: the deleteTissue was not calling the value changed and was not communicating with backend

**Review by GiCharkviani — APPROVED** [review]

**Review by zyme — APPROVED** [review]
LGTM

**Pull request merged**